sources/bloodhound/src/bin/kubernetes-checks/checks.rs

```diff
@@ -0,0 +1,724 @@
+    }
+
+    fn metadata(&self) -> CheckerMetadata {
+        CheckerMetadata {
+            title: "Ensure that the --kubeconfig kubelet.conf file permissions are set to 644 or more restrictive".to_string(),
+            id: "4.1.5".to_string(),
+            level: 1,
+            name: "k8s04010500".to_string(),
+            mode: Mode::Automatic,
+        }
+    }
+}
+
+// =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<=
+
+pub struct K8S04010600Checker {}
+
+impl Checker for K8S04010600Checker {
+    fn execute(&self) -> CheckerResult {
+        ensure_file_owner_and_group_root(KUBELET_KUBECONFIG_FILE)
+    }
+
+    fn metadata(&self) -> CheckerMetadata {
+        CheckerMetadata {
+            title: "Ensure that the --kubeconfig kubelet.conf file ownership is set to root:root"
+                .to_string(),
+            id: "4.1.6".to_string(),
+            level: 1,
+            name: "k8s04010600".to_string(),
+            mode: Mode::Automatic,
+        }
+    }
+}
+
+// =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<=
+
+pub struct K8S04010700Checker {}
+
+impl Checker for K8S04010700Checker {
+    fn execute(&self) -> CheckerResult {
+        let no_x_xwr_xwr = S_IXUSR | S_IRWXG | S_IRWXO;
+        check_file_not_mode(KUBELET_CLIENT_CA_FILE, no_x_xwr_xwr)
+    }
+
+    fn metadata(&self) -> CheckerMetadata {
+        CheckerMetadata {
+            title: "Ensure that the certificate authorities file permissions are set to 600 or more restrictive".to_string(),
+            id: "4.1.7".to_string(),
+            level: 1,
+            name: "k8s04010700".to_string(),
+            mode: Mode::Automatic,
+        }
+    }
+}
+
+// =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<=
+
+pub struct K8S04010800Checker {}
+
+impl Checker for K8S04010800Checker {
+    fn execute(&self) -> CheckerResult {
+        ensure_file_owner_and_group_root(KUBELET_CLIENT_CA_FILE)
+    }
+
+    fn metadata(&self) -> CheckerMetadata {
+        CheckerMetadata {
+            title:
+                "Ensure that the client certificate authorities file ownership is set to root:root"
+                    .to_string(),
+            id: "4.1.8".to_string(),
+            level: 1,
+            name: "k8s04010800".to_string(),
+            mode: Mode::Automatic,
+        }
+    }
+}
+
+// =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<= =>o.o<=
+
+pub struct K8S04010900Checker {}
+
+impl Checker for K8S04010900Checker {
+    fn execute(&self) -> CheckerResult {
+        let no_x_xwr_xwr = S_IXUSR | S_IRWXG | S_IRWXO;
+        check_file_not_mode(KUBELET_CONF_FILE, no_x_xwr_xwr)
+    }
+
+    fn metadata(&self) -> CheckerMetadata {
+        CheckerMetadata {
+            title: "If the kubelet config.yaml configuration file is being used validate permissions set to 600 or more restrictive".to_string(),
```

Comment:
It seems like all the `kubelet` systemd services do pass `--config <path>`, so could this be?:

```suggestion
            title: "Ensure --config config.yaml file permissions are set to 600 or more restrictive".to_string(),
```

Reply:
See above, this is the title of the check.